envclasses/__init__.py

```diff
@@ -132,13 +132,24 @@ class InvalidNumberOfElement(LoadEnvError):
     """
 
 
+def _coalesce(typ: Type) -> Type:
+    if not is_optional_type(typ):
+        return typ
+    else:
+        return cast(
+            Type,
+            Union[tuple(tt for tt in get_args(typ, evaluate=True) if not is_optional_type(tt))])
```

Comment:
I am new to `typing.cast`. What does `_coalesce` actually return, the value type of `Optional[T]`? 🤔

Reply:
Yes exactly, `_coalesce` should return the type `T` from `Optional[T]`. Currently l, the rest of the PR only works is `T` is not a complex type.

[typing.cast](https://docs.python.org/3/library/typing.html#typing.cast) is just syntactic sugar to tell the type checker the type of the value. I couldn't get the linter to work without it.